### DISCUSSION TITLE
how to add new additional fee in cart?

### DISCUSSION THREAD

**Discussion by quang-ng** [medusajs/medusa#3949] [Q&A] [answered]
Currently in cart, we have shipping fee and tax. In my case, I need to add one more fee additional  named “services fee” based on the number of products in the cart. Could someone suggest an approach to implement it in MedusaJS?

**Comment by quangnst** [comment]
any update?? Quang, we are implementing it too

**Reply by quang-ng** [reply]
Sorry @quangnst, we're are discussing about it, not done yet. Will post here when we found a solution.

**Comment by srindom** [comment, marked as answer]
Hi @quang-ng - adding custom line items to a cart is possible. You can use this option to support something like a fee. An option would be to have a custom endpoint along the lines of `POST /store/cart/:id/fees`, which does something like this:

```js
      const feeInclTax = await yourLogicToCalculateFee(cartId)
      await lineItemService.create({
        cart_id: cartId,
        title: "Fees",
        has_shipping: true,
        allow_discounts: false,
        unit_price: feeInclTax,
        quantity: 1,
        includes_tax: true
      })
```

The endpoint could then be called from the checkout flow at a suitable time to display the final total to the customer. You may also want to consider adding your own CartCompletionStrategy to add a validation step that ensures the fees have been added to the cart before the order gets processed.

Hope this helps :)

**Reply by quang-ng** [reply]
Thank you very much for your reply. Let me implement and update the result.

**Reply by recdpz** [reply]
Any update I'm implementing this as well would like to make sure I'm doing it right

**Reply by asrudii** [reply]
> Hi @quang-ng - adding custom line items to a cart is possible. You can use this option to support something like a fee. An option would be to have a custom endpoint along the lines of `POST /store/cart/:id/fees`, which does something like this:
> 
> ```js
>       const feeInclTax = await yourLogicToCalculateFee(cartId)
>       await lineItemService.create({
>         cart_id: cartId,
>         title: "Fees",
>         has_shipping: true,
>         allow_discounts: false,
>         unit_price: feeInclTax,
>         quantity: 1,
>         includes_tax: true
>       })
> ```
> 
> The endpoint could then be called from the checkout flow at a suitable time to display the final total to the customer. You may also want to consider adding your own CartCompletionStrategy to add a validation step that ensures the fees have been added to the cart before the order gets processed.
> 
> Hope this helps :)

Hi, how to do this in new version?

**Reply by quang-ng** [reply]
> Any update I'm implementing this as well would like to make sure I'm doing it right

Sorry @recdpz  for delay, yes, it was work.

**Reply by quang-ng** [reply]
Hi @asrudii, our project is closed, and we haven't made any upgrades, so I'm not sure how to proceed with the new version of Medusa.

**Reply by asrudii** [reply]
This one basicaly add item to cart with item name is "fee" right?
and it will be show in item list as default?

**Comment by quang-ng** [comment]
Yes, you are correct
